platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/jasperdb/VirtualLeafRecordSerializer.java

```diff
@@ -252,12 +247,18 @@ public VirtualLeafRecord<K, V> deserialize(final ByteBuffer buffer, final long d
     @Override
     public int serialize(final VirtualLeafRecord<K, V> leafRecord, final SerializableDataOutputStream outputStream)
             throws IOException {
+        final int hashSerializationVersion = (int) (0x000000000000FFFFL & currentVersion);
         final SerializableDataOutputStream serializableDataOutputStream =
                 hasVariableDataSize ? DATA_FILE_OUTPUT_STREAM_THREAD_LOCAL.get().reset() : outputStream;
+
         // put path (data item key)
         serializableDataOutputStream.writeLong(leafRecord.getPath());
-        // put hash
-        serializableDataOutputStream.write(leafRecord.getHash().getValue());
+        if (hashSerializationVersion != 0) {
+            // We need to persist some value to keep the serialization format consistent.
+            // This flow is relevant when we create an instance of VirtualLeafRecordSerializer by deserializing
+            // and then use the same instance to serialize.
+            serializableDataOutputStream.write(NULL_HASH_VALUE);
+        }
```

Comment:
This is questionable... It would be great, if serializers could read data according to its version, but when it comes to writes, always use the latest version format.

Reply:
Yes, I introduced `REMOVED_LEAF_HASHES` class version and custom serialization/deserialization for the previous version.